pkg/pub_integration/lib/src/headless_env.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:puppeteer/puppeteer.dart';
+// ignore: implementation_imports
+import 'package:puppeteer/src/page/page.dart' show ClientError;
+
+class HeadlessEnv {
+  final Directory tempDir;
+  Browser _browser;
+  final clientErrors = <ClientError>[];
+
+  HeadlessEnv({Directory tempDir})
+      : tempDir =
+            tempDir ?? Directory.systemTemp.createTempSync('pub-headless');
+
+  Future _startBrowser() async {
+    if (_browser != null) return;
+    final chromeBin = Platform.environment['CHROME_BIN'];
+    if (chromeBin != null && !File(chromeBin).existsSync()) {
```

Comment:
Can we please do some bare minimum for automatic resolution, I think we can just search `PATH` or use `google-chrome` if we don't need a absolute path.